### ISSUE TITLE
New Check for assuring empty lines between method declarations

### ISSUE THREAD

**Issue by vanniktech** [checkstyle/checkstyle#3832]
I'd like to check that between method declarations a new line is inserted. Given the class below:

**Failing class**

```java
public class Test {
  public void foo() {

  }
  public void abc() {

  }
}
```

I'd expect the check to raise a warning since a new line should be inserted between the `foo` and the `abc` method.

**Non-Failing class**

```java
public class Test {
  public void foo() {

  }

  public void abc() {

  }
}
```

I haven't found any check to do this currently. Are you open adding a new check for this one?

**Comment by rnveach** [comment]
Try EmptyLineSeparatorCheck.
http://checkstyle.sourceforge.net/config_whitespace.html#EmptyLineSeparator

````
$ cat TestClass.java
public class Test {
  public void foo() {

  }
  public void abc() {

  }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="EmptyLineSeparator"/>
    </module>
</module>

$ java -jar checkstyle-7.5.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:5: 'METHOD_DEF' should be separated from previous statement. [EmptyLineSeparator]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by Sid481** [comment]
@rnveach - thanks for the post, here I would like to know if there is any module available to check if there should not be more than one empty line between two method definition.

**Comment by pbludov** [comment]
> @rnveach - thanks for the post, here I would like to know if there is any module available to check if there should not be more than one empty line between two method definition.

It is possible to completelly forbid consecutive empty lines with regexp: 
https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L75

**Comment by rnveach** [comment]
@Sid481 Also check out http://checkstyle.sourceforge.net/config_whitespace.html#EmptyLineSeparator
